src/client/render/render-core.js

```diff
@@ -148,6 +148,54 @@ let renderCore = function() {
 			}
 		},
 
+		setCameraStyle: function(type) {
+			// If this function is called from an event, use the event origin node to determine what type to become
+			if (this && typeof type !== "string") type = this.dataset.type;
```

Comment:
I think it's best to prevent the use of `data-*`. They're easy to forget when code and/or parameters change, and it requires additional code like this here.
It would be cleaner to call `this.setCameraStyle("freeCamera")` directly in the event listeners above.

Reply:
Agreed, but I'm not sure how to get the DOM node otherwise if I don't change the node ID to the exact type.

Reply:
There's a switch case that could be used to set a local string variable, that it'll use to update the correct node for afterwards.